src/components/Auth/AddTodo/AddTodo.tsx

```diff
@@ -0,0 +1,179 @@
+    setToggleLoader,
+  } = useContext(TodoContext);
+
+  const newTodoField = useRef<HTMLInputElement>(null);
+  const [todoField, setTodoField] = useState('');
+  const [tempTodoCount, setTempTodoCount] = useState<number>(0);
+  const [activeTodoField, setActiveTodoField] = useState(true);
+
+  useEffect(() => {
+    if (newTodoField.current) {
+      newTodoField.current.focus();
+    }
+  }, []);
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      try {
+        if (userId) {
+          const todoData = await getTodos(userId);
+
+          setTodos(todoData);
+          handleFilter(filterState, todoData);
+        }
+      } catch (_) {
+        setLoadError(true);
+        setErrorMessage('Unable to load todos from server');
+      }
+    };
+
+    loadTodos();
+  }, [tempTodoCount]);
+
+  useEffect(() => {
+    if (newTodoField.current && activeTodoField) {
+      newTodoField.current.focus();
+    }
+  }, [activeTodoField]);
+
+  const addTodoToTheServer = async () => {
+    try {
+      if (userId) {
+        const last = tempTodoCount + 1;
+
+        const newTodo = {
+          userId,
+          title: todoField,
+          completed: false,
+
+        };
+
+        await addTodo(userId, newTodo);
+        setTempTodoCount(last);
+      }
+    } catch (_) {
+      setLoadError(true);
+      setErrorMessage('Unable to add new todo to the server');
+      handleFilter(filterState, todos);
+      setTodos(todos);
+    } finally {
+      setTodoField('');
+      setActiveTodoField(true);
+    }
+  };
+
+  const addTodoToTheList = () => {
+    if (userId) {
+      const newTodo = {
+        id: 0,
+        userId,
+        title: todoField,
+        completed: false,
+
+      };
+
+      setTodos([...todos, newTodo]);
+      handleFilter(filterState, [...todos, newTodo]);
+    }
+  };
+
+  const handleNewTodo = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (todoField.trim() !== '') {
+      setActiveTodoField(false);
+      addTodoToTheList();
+      addTodoToTheServer();
+    } else {
+      setErrorMessage('Empty title is not valid');
+      setLoadError(true);
+    }
+  };
+
+  const handleInputField = (event: ChangeEvent<HTMLInputElement>) => {
+    setTodoField(event.target.value);
+  };
+
+  const toggleAllOnTheServer = async () => {
+    setToggleLoader(true);
+    await Promise.all(
+      todos.map(async (item) => {
+        const completeTodo = { ...item };
+
+        if (todos.some(item2 => !item2.completed)) {
+          completeTodo.completed = true;
+        } else {
+          completeTodo.completed = false;
+        }
+
+        await updateTodo(item.id, completeTodo);
+      }),
+    );
+
+    const newTodo = todos.map(item => {
+      const toggleTodo = { ...item };
+
+      if (todos.some(item2 => !item2.completed)) {
+        toggleTodo.completed = true;
+
+        return toggleTodo;
+      }
+
+      toggleTodo.completed = false;
+
+      return toggleTodo;
+    });
```

Comment:
Now if we get an error on the response from the server, we anyway update todos local. Need show error if something went wrong.

You can create a method that will send the request to the server. If the response from the server will be successful (server returns updated todo), you need to update the current todo on the state. If not, show an error. And you can call this function for each complete todo

Reply:
I tried to fix this and create separate method for updating todos for 2 days but every time i just broke my function and localy todos just stopped being displayed